### ISSUE TITLE
State Broken in Main Event Filter

### ISSUE THREAD

**Issue by HansVraga** [USGS-WiM/STNWeb#427]
From Todd:
>I just noticed that the EVENT STATE filter in the main Choose Event dialog doesn't do anything.
![image](https://user-images.githubusercontent.com/12058013/38941715-2b8ef9b0-42fb-11e8-93ea-17bdad64eca2.png)

**Comment by HansVraga** [comment]
I did a bit of troubleshooting myself - clicking "Filter Events" seemed to work sometimes, and sometimes not, and one of the filters didn't make much sense...

I think we might have been dealing with a delay in the filter - if that is the case perhaps we need to show a spinner?

**Comment by lprivette** [comment]
The `FilteredEvents` endpoint is made by joining the `instrument` table (has event_id & site_id) and then the `sites` table in order to get the states for each event. 

That's a ton of records being returned which is why it looks like nothing is happening in the client-- it's taking a long time to perform the query and then fails. You can get results if you specify a relatively recent start time:
![image](https://user-images.githubusercontent.com/23640741/53363237-f3d7b600-3909-11e9-9a07-023536f6dbff.png)

Options I see going forward:

1. Remove it 
2. Restrict event start date to be within the last year or 2 and throw in a loading spinner. (bleh)
3. Refactor the services so that the state is either in the events table or more easily accessible.

Might also want to run this by Jeremy in case I'm missing something.

**Comment by HansVraga** [comment]
How long is a long time?

**Comment by lprivette** [comment]
Got a working fix for it in this branch https://github.com/USGS-WiM/STNServices2/tree/filteredEventsFix

I think we should do this in Test before putting putting in Prod. 

What do you think about doing the clone of prod and making that the new STNTEST? We'd then leave the existing STNTEST for hooking up the new services. That way we're giving Todd and co their testing environment back.

**Comment by lprivette** [comment]
- [ ] check FEV to see if that endpoint is used

**Comment by lprivette** [comment]
in test

**Comment by tkoenigusgs** [comment]
This is much better.  It's fast!   For States that produce a smaller event list, I could typically make my selection (e.g. Texas), then click the "Filter Events" button, then wait just a couple seconds, then click the event drop down and see the new list.  However, if I moved pretty quickly from the Filter Events button to the Events drop-down, I would often still see my previous list, and I would have to click and re-click the Events drop-down until it showed me the new list.  So... I think the typical user might not notice the delay if they are not clicking as fast as I am.  But **a loading spinner on the Filter Events button might be necessary here to slow down the more seasoned users**.  On my particular system, I did different States over and over again and found that Texas needs a 2 second delay while Iowa and Pennsylvania need 4 seconds to load.
